### PR TITLE
feat(alerts): New empty / welcome state messages

### PR DESCRIPTION
**Pull request by evanpurkhiser**
* Adds a new 'welcome' state that will dismiss after the user sees it
   for the first time

 * Updates the empty state to be determanistic on if they've already
   configured alert rules or not

### CODE REVIEW

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx [hunk 1]
```diff
@@ -110,30 +136,60 @@ class IncidentsList extends AsyncComponent<Props, State & AsyncComponent['state'
...
+        leftAligned
+        size="medium"
+        title={t('Find the signal in the noise')}
+        description={t(
+          'You’ve got 5 minutes, 2 million lines of code, and an inbox with 300 new messages. Alerts tell you what went wrong and why.'
+        )}
+        action={
+          <ButtonBar gap={1}>
+            <Button size="small" external href={DOCS_URL}>
+              View Features
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/alerts/list/index.tsx:157]:
```suggestion
              {t('View Features')}
```

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx [hunk 2]
```diff
@@ -338,6 +385,32 @@ class IncidentsListContainer extends React.Component<Props> {
...
+            ? t('Users with admin permission or higher can create alert rules.')
+            : undefined
+        }
+        onClick={e => {
+          e.preventDefault();
+
+          navigateTo(`/settings/${params.orgId}/projects/:projectId/alerts/new/`, router);
+        }}
+        priority="primary"
+        href="#"
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/alerts/list/index.tsx:404]: Do we need to specify this?

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx [hunk 3]
```diff
@@ -110,30 +136,60 @@ class IncidentsList extends AsyncComponent<Props, State & AsyncComponent['state'
...
+    const {firstVisitShown} = this.state;
+
+    if (!firstVisitShown) {
+      return null;
+    }
+
+    return (
+      <WelcomeEmptyMessage
+        leftAligned
+        size="medium"
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/alerts/list/index.tsx:149]: I noticed that we use "sm, md, lg....." maybe you should rename it to `md` :)

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx (hunk 1 above)
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/alerts/list/index.tsx:157]:
```suggestion
             {t(' View Features')}
```

File: src/sentry/static/sentry/app/views/settings/components/emptyMessage.tsx [hunk 4]
```diff
@@ -33,11 +43,20 @@ const EmptyMessage = styled(
   )
 )<WrapperProps>`
   display: flex;
-  text-align: center;
-  align-items: center;
+  ${p =>
+    p.leftAligned
+      ? css`
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/components/emptyMessage.tsx:48]: I also was using css to have a nice alignment, but after getting some feedback  I stopped using it ..are you sure you want to use them here?

**Review by matejminar — APPROVED** [review]
Nice 👍

Looks like there are some misbehaved tests/snapshots 🔪

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx [hunk 5]
```diff
@@ -28,20 +28,25 @@ import withOrganization from 'app/utils/withOrganization';
...
+import {promptsUpdate} from 'app/actionCreators/prompts';
 
 import {Incident} from '../types';
 import {TableLayout, TitleAndSparkLine} from './styles';
 import AlertListRow from './row';
 
 const DEFAULT_QUERY_STATUS = 'open';
 
+const DOCS_URL =
+  'https://docs.sentry.io/workflow/alerts-notifications/alerts/?_ga=2.21848383.580096147.1592364314-1444595810.1582160976';
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/alerts/list/index.tsx:40]: What is this querystring for?

File: src/sentry/static/sentry/app/views/settings/components/emptyMessage.tsx (hunk 4 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/settings/components/emptyMessage.tsx:48]: Using `css` tagged strings inside a template string doesn't do anything which is why I've been suggesting we don't do it.

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx (hunk 5 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/alerts/list/index.tsx:40]: Not sure, copied from where it was before. @adhiraj ?

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx (hunk 3 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/alerts/list/index.tsx:149]: Will do in a separate PR :)

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx (hunk 5 above)
Comment (adhiraj) [reply src/sentry/static/sentry/app/views/alerts/list/index.tsx:40]: Not sure, we can remove it

**Pull request merged**